test/org/loklak/harvester/RedirectUnshortenerTest.java

```diff
@@ -21,7 +21,7 @@ public void testRedirectUnshortener() {
         shortlinkMap.put("http://ow.ly/P9SX30ddSTI", "https://github.com/loklak/loklak_server/issues/1284");
         shortlinkMap.put("http://bit.do/blog-fossasia", "http://blog.fossasia.org/");
         shortlinkMap.put("http://fb.me/4lcXZsyyO", "https://www.facebook.com/permalink.php?story_fbid=1550262581900846&id=1381813572079082");
-        shortlinkMap.put("http://wp.me/sf2B5-shorten", "https://en.blog.wordpress.com/2009/08/14/shorten/");
+        shortlinkMap.put("http://wp.me/sf2B5-shorten", "http://en.blog.wordpress.com/2009/08/14/shorten/");
```

Comment:
> My question is why does this test have `http://` instead of `https://` when `https://` works?

To explain this, let me take an example - 
`http://bit.ly/xyz` -> `http://something.new` -> `https://something.new`

Now, the URL unshortener would request `http://bit.ly/xyz` and get a `30X` response code asking for a redirect to `http://something.new`. Now, this URL is not `isApplicable()`, so the unshortener would not request this URL (and hence wouldn't get redirected to `https://something.new`). And when a user requests for `http://something.new` (after getting the URL from loklak's API), his browser redirects to `https://something.new`.

This would save loklak at least one HTTP request (maybe `https://something.new` -> `http://something.newer` -> `https://something.newer`) before finalizing a Tweet.

> Does the test fail when it's https?

After this change, yes.

> At what stage of URL redirect solution do you decide that the URL is not `anonymized`.

If it `isApplicable()`, it needs to be unshortened.


Just to mention, the [previous version of `RedirectUnshortener`](https://github.com/loklak/loklak_server/blob/ef587d75cf40c258e7996454c2a1838e599b618e/src/org/loklak/harvester/RedirectUnshortener.java#L80) also used this logic.